test/functional/rpc_getdescriptoractivity.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from decimal import Decimal
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_raises_rpc_error, assert_is_hex_string
+from test_framework.messages import COIN
+from test_framework.wallet import MiniWallet, getnewdestination
+
+
+class GetBlocksActivityTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def run_test(self):
+        node = self.nodes[0]
+        wallet = MiniWallet(node)
+        node.setmocktime(node.getblockheader(node.getbestblockhash())['time'])
+        wallet.generate(200, invalid_call=False)
+
+        self.test_no_activity(node)
+        self.test_activity_in_block(node, wallet)
+        self.test_no_mempool_inclusion(node, wallet)
+        self.test_multiple_addresses(node, wallet)
+        self.test_invalid_blockhash(node, wallet)
+        self.test_invalid_descriptor(node, wallet)
+        self.test_confirmed_and_unconfirmed(node, wallet)
+        self.test_receive_then_spend(node, wallet)
+
+    def test_no_activity(self, node):
+        _, spk_1, addr_1 = getnewdestination()
+        result = node.getdescriptoractivity([], [f"addr({addr_1})"], True)
+        assert_equal(len(result['activity']), 0)
+
+    def test_activity_in_block(self, node, wallet):
+        _, spk_1, addr_1 = getnewdestination()
+        txid = wallet.send_to(from_node=node, scriptPubKey=spk_1, amount=1 * COIN)['txid']
+        blockhash = self.generate(node, 1)[0]
+
+        # Test getdescriptoractivity with the specific blockhash
+        result = node.getdescriptoractivity([blockhash], [f"addr({addr_1})"], True)
+        assert_equal(list(result.keys()), ['activity'])
+        [activity] = result['activity']
+
+        for k, v in {
+                'amount': Decimal('1.00000000'),
+                'blockhash': blockhash,
+                'height': 201,
+                'txid': txid,
+                'type': 'receive',
+                'vout': 1,
+        }.items():
+            assert_equal(activity[k], v)
+
+        outspk = activity['output_spk']
+
+        assert_equal(outspk['asm'][:2], '1 ')
+        assert_equal(outspk['desc'].split('(')[0], 'rawtr')
+        assert_is_hex_string(outspk['hex'])
```

Comment:
In dd19d076c528a075d02fa66aaf906f96fa314450 "rpc: add getdescriptoractivity"

The scriptPubKey bytes are known, so this can directly check they are the same:

```suggestion
        assert_equal(outspk['hex'], spk_1.hex())
```